pkgs/development/web/nhost/default.nix

```
@@ -0,0 +1,29 @@
+{ buildGoModule
+, fetchFromGitHub
+, lib
+}:
```

```suggestion
{ 
  buildGoModule
, fetchFromGitHub
, lib
}:
```

This makes diffs more readable.

---

done, I just copied another example

---

@cafkafk Do you use a tool to format the file? When using `nixpkgs-fmt`, I have the original PR formatting and this is the one that I usually recommend. It would be nice if reviewers could be on the same wavelength on this or else contributors might go crazy.

---

> It would be nice if reviewers could be on the same wavelength on this or else contributors might go crazy.

Agreed!

The style I'm using is based on what's been recommended to me on reviews I've gotten, I tried using nixpkgs-fmt, but reviewers nearly always had problems with it. 

I'm only recommending this because it's what my experience suggests is idiomatic, and what I would expect other reviewers would mention if I didn't. I may be wrong however!

It is a problem that there isn't an unified style guide for nixpkgs, but what I'm arguing for aligns more with one popular style guide from e.g. [alejandra](https://github.com/kamadorueda/alejandra) which quite nicely argues for this.

Allow me to copy-paste it here:

# from alejandra style guide

✅ Good:

```nix
{
  mkDerivation,
  lib,
  fetchurl,
  ...
} @ attrs:
  stdenv.mkDerivation # ...
```

- Adding an argument produces a minimal diff
  (including the first and last elements):

  ```patch
    mkDerivation,
    lib,
    fetchurl,
  + google-chrome-stable,
  ```

- Removing an argument produces a minimal diff
  (including the first and last elements):

  ```patch
    mkDerivation,
  - lib,
    fetchurl,
  ```

- The comma at the end is consistent with _Let-In_, and _Map_,
  where the separator goes after the element
  instead of at the beginning.

❌ Bad:

<!-- nixpkgs-fmt -->

```nix
{ lib
, mkDerivation
, fetchurl
, ...
} @ attrs:
stdenv.mkDerivation # ...
```

- Removing the first element
  produces a diff in two elements:

  ```diff
  - { lib
  - , mkDerivation
  + { mkDerivation
    , fetchurl
    , ...
    } @ attrs:
    stdenv.mkDerivation # ...
  ```

- Documenting the first argument creates an inconsistency
  between the way arguments start:

  ```nix
  {
    # Lorem Ipsum
    lib
  , mkDerivation
  , fetchurl
  , ...
  } @ attrs:
  stdenv.mkDerivation # ...
  ```

- This is not consistent with _Let-In_, and _Map_,
  where the separator goes after the element
  instead of at the beginning.
- It ruins "folding by indentation" modes
  on Vim, Neovim, VSCode, and other major code editors,
  because the data-structure has the same indentation
  as the opening bracket.

❌ Bad:

<!-- nixfmt -->

```nix
{ mkDerivation, lib, fetchurl, ... }@attrs: stdenv.mkDerivation # ...
```

- One-liners are unreadable.

❌ Bad:

<!-- nixfmt -->

```nix
{ mkDerivation, lib, fetchurl, extra-cmake-modules, kdoctools, wrapGAppsHook
, karchive, kconfig, kcrash, kguiaddons, kinit, kparts, kwind, ... }@attrs:
stdenv.mkDerivation # ...
```

- It's hard to tell this destructuring has an ellipsis (`...`) at a first glance,
  because it's mixed with the other arguments.
- Moving elements becomes harder
  than a simple whole-line movement.
  (Moving a whole line is normally a keyboard-shortcut
  or command in major code editors).
- Excessively compact:
  adding, removing, or editing an argument
  produces a diff in more than one argument.
- `}@attrs` is not intuitive
  with the rules of written english,
  where you add whitespace
  after the end of the previous phrase
  (`phrase. Other phrase`).

---

Of course either way isn't blocking to me, it's more an improvement suggestion, whatever way @andr-ec goes with that pleases you is fine with me!

---

Impressive. Maybe I should use your syntax. I'll think about it. Thanks for the detailed explanation !!!

---

> Removing the first element produces a diff in two elements:

`lib` and `stdenv` were always the first elements. Then, at least in Nixpkgs, this particular case never happens.

And I believe this is too ugly and injustifiable for the input attrset.